tasks/native_gem.rake

```diff
@@ -5,10 +5,10 @@ task 'gem:windows' => ['ports:cross'] do
   require 'rake_compiler_dock'
 
   # make sure to install our bundle
-  build = ['bundle']
+  sh "bundle package --all"   # Avoid repeated downloads of gems by using gem files from the host.
 
   # and finally build the native gem
-  build << 'rake cross native gem RUBY_CC_VERSION=2.7.0:2.6.0:2.5.0:2.4.0 CFLAGS="-Wall" MAKE="make -j`nproc`"'
-
-  RakeCompilerDock.sh build.join(' && ')
+  GEM_PLATFORM_HOSTS.keys.each do |plat|
+    RakeCompilerDock.sh "bundle --local && RUBY_CC_VERSION='2.7.0:2.6.0:2.5.0:2.4.0' CFLAGS='-Wall' MAKE='make -j`nproc`' rake native:#{plat} gem", platform: plat
```

Comment:
1. This command switched from `rake cross native gem` to `rake native:{plat}`. Why does this need to change?
2. I see we have Ruby 2.4 on the cross compile list. Given the other commit about bundler (etc), and that Ruby 2.4 is well out of support, what do you think about dropping it? That'd mean that our current release is the last one tested on 2.4. We have the same questions about 2.5 and 2.6, although those ought to continue to work for the time being, so I don't think we also need to drop those right now.